### PR TITLE
Passing ModuleConfig rather than ConfigService

### PR DESCRIPTION
**Pull request by sendilkumarn**
Ideally both are same, but useFactory is well supported by AOT

### CODE REVIEW

**Comment by sendilkumarn** [comment]
@deepu105  on hold

**Comment by sendilkumarn** [comment]
This will fix the error that we faced in aot compilation.

**Comment by sendilkumarn** [comment]
@pascalgrimaud Removed `on-hold` now.  👍

File: src/config.ts [hunk 1]
```diff
@@ -1,25 +1,28 @@
 /*
  * Copyright 2016 the original author or authors.
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
+ * Licensed under the Apache License; Version 2.0 (the "License");
```
Comment (jdubois) [inline src/config.ts]: You have some issue with the license here, but this is a more general problem, @deepu105 we need to have the same license everywhere, I'll send an email to everyone about this

**Comment by sendilkumarn** [comment]
That remains the same. Since there is no change in retrieving value from configservice

File: src/config.service.ts [hunk 2]
```diff
@@ -15,29 +15,30 @@
  */
 import { Injectable } from '@angular/core';
 import { ModuleConfig } from './config';
-import { CONSTANTS } from './constants';
 
 @Injectable()
 export class ConfigService {
     CONFIG_OPTIONS: ModuleConfig;
+    defaultConfig = new ModuleConfig();
```
Comment (deepu105) [inline src/config.service.ts]: Can we skip this line and assign new value directly in constructor

File: src/config.ts (hunk 1 above)
Comment (deepu105) [inline src/config.ts]: Sendil can you remove the changes from the licence here

**Comment by deepu105** [comment]
@sendilkumarn @flaviencathala I'll do a new version release for this. Can you update the generator and see if AOT works?

**Comment by sendilkumarn** [comment]
@deepu105 waiting for a new release.

**Comment by sendilkumarn** [comment]
AOT compilation works 😉 😆

**Comment by deepu105** [comment]
@sendilkumarn I have released a new version with NG4 update. Please checks again with v0.1.11

**Pull request merged**